tools/src/main/java/org/apache/kafka/trogdor/workload/ExternalCommandWorker.java

```diff
@@ -0,0 +1,227 @@
+        this.doneFuture = doneFuture;
+        executor.submit(new Prepare());
+    }
+
+    public class Prepare implements Runnable {
+        @Override
+        public void run() {
+            try {
+                ArrayList<String> command = new ArrayList<String>(spec.command());
+                command.add("--spec");
+                command.add(spec.toString());
+                log.info("ExternalCommandWorker executes command: {}", command);
+                ProcessBuilder workerBuilder = new ProcessBuilder(command);
+                worker = workerBuilder.start();
+                Future updater = executor.submit(new StatusUpdater());
+                Future errorUpdater = executor.submit(new ErrorUpdater());
+                worker.waitFor();
+                updater.get();
+                errorUpdater.get();
+                int workerExitValue = worker.exitValue();
+                log.info("ExternalWorker finished with the exit value: {}.", workerExitValue);
+                if (workerExitValue != 0 && errMsg.isEmpty()) {
+                    errMsg = "ExternalWorker exited with error code " + worker.exitValue();
+                }
+                log.info("StatusUpdater terminated.");
+                doneFuture.complete(errMsg);
+            } catch (IOException e) {
+                log.info("Stdout of ExternalWorker is closed.");
+            } catch (InterruptedException e) {
+                log.info("StatusUpdadter interrupted.");
+            } catch (Exception e) {
+                WorkerUtils.abort(log, "Prepare", e, doneFuture);
+            }
+        }
+    }
+
+    public class ErrorUpdater implements Runnable {
+        BufferedReader br;
+        ErrorUpdater() {
+            br = new BufferedReader(new InputStreamReader(worker.getErrorStream(), StandardCharsets.UTF_8));
+        }
+        @Override
+        public void run() {
+            try {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    log.error("ExternalWorker (stderr):{}", line);
+                    errMsg = line;
+                }
+            } catch (IOException ioe) {
+                log.info("Stderr of ExternalWorker is closed.");
+            } catch (Exception e) {
+                WorkerUtils.abort(log, "ErrorUpdater", e, doneFuture);
+            }
+        }
+    }
+
+    /**
+     * StatusUpdater reads ProcessWorker's stdout line by line. If the line is a JSON object, StatusUpdater updates
+     * the status with the JSON object.
+     */
+    public class StatusUpdater implements Runnable {
+
+        BufferedReader br;
+        StatusUpdater() {
+            br = new BufferedReader(new InputStreamReader(worker.getInputStream(), StandardCharsets.UTF_8));
+        }
+
+        @Override
+        public void run() {
+            try {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    log.info("Worker (stdout):{}", line);
```

Comment:
Can we put the worker ID into the prefix here? We may be running multiple workers at once.